### PR TITLE
chore: Add test for booking cancelled webhook

### LINKED ISSUE #14886: [CAL-3641] Add test for booking cancelled webhook
### Is your proposal related to a problem?

Add test for https://github.com/calcom/cal.com/pull/14745/files

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

<sub>[CAL-3641](https://linear.app/calcom/issue/CAL-3641/add-test-for-booking-cancelled-webhook)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

Fixes #14886 
Fixes CAL-3641

## Type of change

- Tests (Unit/Integration/E2E or any other test)

## How should this be tested?

`yarn test cancel.test.ts
`

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
@Amit91848 we wanted unit tests here. one of the contributor created this function expectBookingCancelledWebhookToHaveBeenFired (https://github.com/calcom/cal.com/pull/14745/files) which we can use.

File: apps/web/playwright/webhook.e2e.ts [hunk 1]
```diff
@@ -164,6 +164,121 @@ test.describe("BOOKING_CREATED", async () => {
   });
 });
 
+test.describe("BOOKING_CANCELLED", async () => {
```
Comment (Udit-takkar) [inline apps/web/playwright/webhook.e2e.ts:167]: can you remove this E2E test?

File: packages/features/bookings/lib/handleNewBooking/test/cancel.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        },
+      },
+    });
+
+    const { req: mockCreateBookingReq } = createMockNextJsRequest({
+      method: "POST",
+      body: mockBookingData,
+    });
+    const createdBooking = await handleNewBooking(mockCreateBookingReq);
+
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/test/cancel.test.ts:98]: you need to add some assertions to make sure booking was created successfully.

Comment (Udit-takkar) [reply packages/features/bookings/lib/handleNewBooking/test/cancel.test.ts:98]: you can check other unit test

**Comment by Amit91848** [comment]
Unit test passes but there is an error thrown when cancelling scheduled jobs

When booking is created scheduledJobs is null
https://github.com/Amit91848/cal.com/blob/87a260eba39c059f2626d6257a0d2dcb79f05986/packages/features/bookings/lib/handleNewBooking/test/cancel.test.ts#L101

Screenshot 2024-05-14 at 12 37 10 PM


But when you fetch it from database it converts to an object?? Like here
https://github.com/Amit91848/cal.com/blob/87a260eba39c059f2626d6257a0d2dcb79f05986/packages/features/bookings/lib/handleNewBooking/test/cancel.test.ts#L112-L118
Screenshot 2024-05-14 at 12 38 28 PM

File: packages/features/bookings/lib/handleCancelBooking/test/webhook.test.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      organizer,
+      location: BookingLocations.CalVideo,
+      subscriberUrl: "http://my-webhook.example.com",
+      payload: {
+        organizer: {
+          id: organizer.id,
+          username: organizer.username,
+          email: organizer.email,
+          name: organizer.name,
+          timeZone: organizer.timeZone,
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleCancelBooking/test/webhook.test.ts:128]: I added this new test instead. 

we just wanted to test booking cancelled so i just created a dummy booking using `getScenarioData` and also you skipped checking if organizer id is passed in payload which is the main thing that needs to be checked

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**